src/renderer/atlas/BackendD3D.cpp

```diff
@@ -2042,10 +2042,11 @@ void BackendD3D::_debugShowDirty(const RenderingPayload& p)
 
     for (size_t i = 0; i < std::size(_presentRects); ++i)
     {
-        if (const auto& rect = _presentRects[i])
+        const auto& rect = _presentRects[(_presentRectsPos + i) % std::size(_presentRects)];
+        if (rect.non_empty())

```

Comment:
This makes debugging with `ATLAS_DEBUG_SHOW_DIRTY` a bit better. It's compiled out in release builds.